### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by NicolasDorier** [btcpayserver/btcpayserver#4231]
### What is your BTCPay version?

1.6.12

### How did you deploy BTCPay Server?

docker

### What happened?

Errors on the swagger schema reported by user.
```
Errors: 
        -paths.'/api/v1/stores/{storeId}/custodian-accounts'. Declared path parameter storeId needs to be defined as a path parameter in path or operation level
        -attribute components.schemas.CustodianData.items is missing
        -paths.'/api/v1/stores/{storeId}/custodian-accounts/{accountId}'. Declared path parameter accountId needs to be defined as a path parameter in path or operation level
        -paths.'/api/v1/stores/{storeId}/users'. Declared path parameter storeId needs to be defined as a path parameter in path or operation level
        -paths.'/api/v1/stores/{storeId}/custodian-accounts/{accountId}'. Declared path parameter storeId needs to be defined as a path parameter in path or operation level
Warnings: 
        -paths.'/api/v1/stores/{storeId}/custodian-accounts'. Declared path parameter storeId needs to be defined as a path parameter in path or operation level
        -attribute components.schemas.CustodianData.items is missing
        -paths.'/api/v1/stores/{storeId}/custodian-accounts/{accountId}'. Declared path parameter accountId needs to be defined as a path parameter in path or operation level
        -paths.'/api/v1/stores/{storeId}/users'. Declared path parameter storeId needs to be defined as a path parameter in path or operation level
        -paths.'/api/v1/stores/{storeId}/custodian-accounts/{accountId}'. Declared path parameter storeId needs to be defined as a path parameter in path or operation level
```
this is using maven's plugin openapi-generator-maven-plugin

### How did you encounter this bug?

using https://docs.btcpayserver.org/API/Greenfield/v1/swagger.json

### Are you sure this is a bug report?

- [X] I confirm this is a bug report

**Comment by NicolasDorier** [comment]
It might be caused by https://docs.btcpayserver.org/API/Greenfield/v1/swagger.json not being in sync with the latest version from
https://mainnet.demo.btcpayserver.org/swagger/v1/swagger.json

Ping @dennisreimann

**Comment by NicolasDorier** [comment]
on mainnet, user reported 
```
Errors: 
        -attribute components.schemas.CustodianData.items is missing
        -paths.'/api/v1/stores/{storeId}/users'. Declared path parameter storeId needs to be defined as a path parameter in path or operation level
Warnings: 
        -attribute components.schemas.CustodianData.items is missing
        -paths.'/api/v1/stores/{storeId}/users'. Declared path parameter storeId needs to be defined as a path parameter in path or operation level
```

**Comment by dennisreimann** [comment]
The docs not being in sync has been fixed in btcpayserver/btcpayserver-doc@4fb2badcf4fcb7be4160beb16158543705c54c0e.

@woutersamaey will take a look at the validation problems.

**Comment by woutersamaey** [comment]
I checked the file on `master` by downloading the `swagger.json` and running a validation tool.
```
docker run --rm -v `pwd`/swagger.json:/swagger.json jeanberu/swagger-cli swagger-cli validate swagger.json
```
Output is
```
swagger.json is valid
```
This validator uses swagger-cli. Not sure it this is the reference or not, but it looks official to me.
It uses this NPM package: https://www.npmjs.com/package/swagger-cli

**Comment by Kukks** [comment]
Keep in mind there are multiple version of swagger, the newest one is
called openapi 3.0

**Comment by woutersamaey** [comment]
@Kukks this validation tool is reporting version `4.0.4` when running `docker run --rm jeanberu/swagger-cli swagger-cli -v`

**Comment by nmpg123** [comment]
Hi. I get the errors when generating Java code using openapi-generator-maven-plugin version 6.2.0.

**Comment by woutersamaey** [comment]
@nmpg123 Can you explain me how to run this (quickly and easily) ?

**Comment by dennisreimann** [comment]
@Kukks @woutersamaey Please also see the two PRs I've referenced.

**Comment by nmpg123** [comment]
@woutersamaey You can use as a Maven plugin as described here: https://stackoverflow.com/questions/70244754/is-it-possible-to-use-the-openapi-generator-maven-plugin-to-create-a-java-client

regular maven build command will trigger the generation e.g: mvn package

**Comment by nmpg123** [comment]
I think there might be an issue with the maven plugin itself tbh. It can be used through an API as follows:

curl -X POST -H "content-type:application/json" \
  -d '{"openAPIUrl":"https://docs.btcpayserver.org/API/Greenfield/v1/swagger.json"}' \
  http://api.openapi-generator.tech/api/gen/clients/java

The generated maven project fails at build with syntax errors e.g.

[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.624 s
[INFO] Finished at: 2022-10-25T15:28:41+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project openapi-java-client: Compilation failure: Compilation failure: 
[ERROR] /home/averagejoe/Downloads/java-client/src/main/java/org/openapitools/client/model/GenericPaymentMethodDataData.java:[75,33] error: > expected
[ERROR] /home/averagejoe/Downloads/java-client/src/main/java/org/openapitools/client/model/GenericPaymentMethodDataData.java:[75,34] error: not a statement
[ERROR] /home/averagejoe/Downloads/java-client/src/main/java/org/openapitools/client/model/GenericPaymentMethodDataData.java:[75,37] error: not a statement

I guess I will be implementing the API myself.

**Comment by pavlenex** [comment]
@dennisreimann @woutersamaey Is this issue fixed by #1117?

**Comment by dennisreimann** [comment]
Yes, that and #4235 resolved it. Closing it for now, we can open another one if there are still things to fix.
